modules/nextflow/src/main/groovy/nextflow/exception/ProcessStageException.groovy

```diff
@@ -24,5 +24,5 @@ import groovy.transform.InheritConstructors
  * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
  */
 @InheritConstructors
-class ProcessStageException extends ProcessException implements ShowOnlyExceptionMessage {
+class ProcessStageException extends ProcessUnrecoverableException implements ShowOnlyExceptionMessage {
```

Comment:
What's the rationale to change this to "unrecoverable" ?

Reply:
It is related to #5727. When there is a problem with data staging, nextflow retries the execution but hash is null. ProcessStageException is changed to unrecoverable to avoid this task retries.

Reply:
Done